astropy/io/fits/convenience.py

```diff
@@ -451,6 +452,10 @@ def writeto(
         If `True`, adds both ``DATASUM`` and ``CHECKSUM`` cards to the
         headers of all HDU's written to the file.
     """
+    if filename is sys.stdout:
+        # special case stdout for debugging convenience
+        # see https://github.com/astropy/astropy/issues/3427
+        filename = filename.buffer
```

Comment:
this one is probably useless since it will go through `HDUList.writeto` isn't it ?

Reply:
indeed, nice catch ! I'll remove it.